### PR TITLE
DOC: fix EX03 Errors for pandas.DataFrame.to_sql

### PR DESCRIPTION
**Pull request by erichxchen**
- [X] Ref: https://github.com/pandas-dev/pandas/issues/56804

Fix these EX03 Errors for `pandas.DataFrame.to_sql`:

```
9 Errors found for `pandas.DataFrame.to_sql`:
        flake8 error: line 8, col 4: E111 indentation is not a multiple of 4
        flake8 error: line 15, col 4: E111 indentation is not a multiple of 4
        flake8 error: line 19, col 4: E111 indentation is not a multiple of 4
        flake8 error: line 27, col 1: F821 undefined name 'df_conflict'
        flake8 error: line 27, col 89: E501 line too long (124 > 88 characters)
        flake8 error: line 28, col 1: F811 redefinition of unused 'insert' from line 22
        flake8 error: line 39, col 1: F821 undefined name 'df_conflict'
        flake8 error: line 39, col 89: E501 line too long (123 > 88 characters)
        flake8 error: line 46, col 3: E111 indentation is not a multiple of 4
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3042,7 +3042,7 @@ def to_sql(
         For MySQL, a callable to update columns ``b`` and ``c`` if there's a conflict
         on a primary key.
 
-        >>> from sqlalchemy.dialects.mysql import insert
+        >>> from sqlalchemy.dialects.mysql import insert  # doctest: +SKIP
```
Comment (datapythonista) [inline pandas/core/generic.py:3045]: Nice changes, just one comment. The error seems to suggest that `insert` is already imported before but never used. Can you have a look, and remove the previous import if that's really the case.

Comment (erichxchen) [reply pandas/core/generic.py:3045]: It's actually used in L3036:
```
...     stmt = insert(table.table).values(data).on_conflict_do_nothing(index_elements=["a"])
```
I am not quite sure why it shows this error, and thus I think we could just  `# doctest: +SKIP` this line

**Review by datapythonista — APPROVED** [review]

**Comment by natmokval** [comment]
@erichxchen, thank you for working on this. I think, there are some errors left that need to be fixed:
```
5 Errors found for `pandas.DataFrame.to_sql`:
        EX03    flake8 error: line 27, col 1: F821 undefined name 'df_conflict'
        EX03    flake8 error: line 27, col 89: E501 line too long (124 > 88 characters)
        EX03    flake8 error: line 28, col 1: F811 redefinition of unused 'insert' from line 22
        EX03    flake8 error: line 39, col 1: F821 undefined name 'df_conflict'
        EX03    flake8 error: line 39, col 89: E501 line too long (123 > 88 characters)
```
For example, line 
https://github.com/pandas-dev/pandas/blob/02011f2b81743f4e9d100aea6c791fa5e8de9933/pandas/core/generic.py#L3041
is too long. Could you please rewrite it as multi-line statement? It will fix one of the error.

**Comment by mroeschke** [comment]
Closed by https://github.com/pandas-dev/pandas/pull/57025

**Pull request closed without merging**